pagerduty/resource_pagerduty_extension.go

```diff
@@ -58,6 +67,7 @@ func buildExtensionStruct(d *schema.ResourceData) *pagerduty.Extension {
 			ID:   d.Get("extension_schema").(string),
 		},
 		ExtensionObjects: expandServiceObjects(d.Get("extension_objects")),
+		Config:           expandExtensionConfig(d.Get("config")),
```

Comment:
Since this is an optional field I believe you must check if the field is in the configuration using `GetOk`.

Example:

```go
if v, ok := d.GetOk("config"); ok {
  Extension.Config = v
}
```

Reply:
Indeed, this will avoid misleading unmarshalling errors.
Done.